### ISSUE TITLE
Has anyone tried using ggml metal to get whisper.cpp to do inference on the M1/M2 GPU?

### ISSUE THREAD

**Issue by david-rokeby** [ggml-org/whisper.cpp#1008]
Has anyone tried using ggml metal to get whisper.cpp to do inference on the M1/M2 GPU?

**Comment by colinc** [comment]
No. But, if you modify the Core ML encoder to use MLComputeUnitsCPUAndGPU, there is a significant speed up on certain hardware (i.e. Max/Ultra chipsets). For example, the M1 Ultra sees a 5x improvement vs the default Core ML implementation. However, it comes at the cost of reduced accuracy for some audio diarization (similar results to CPU). I'm not sure why the ANE has better diarization results for certain audio types. But, if you're not using diarization or working around the accuracy issues in alternative ways, it can be a signifiant increase in speed.
